src/main/src/main/java/org/geoserver/catalog/CascadeDeleteVisitor.java

```diff
@@ -134,7 +134,7 @@ public void visit(LayerInfo layer) {
                 
                 // either update or remove the group
                 if(group.getLayers().size() == 0) {
-                    catalog.remove(group);
+                    visit(catalog.getLayerGroup(group.getId()));
```

Comment:
Had a bit of a hunt to find that the remove( group ) method is still being called inside the visit.